### PR TITLE
fix JsonToStringStyle

### PR DESCRIPTION
**Pull request by jacktan1991**
except number, boolean, Array and nested object structures, other types like char, String, Date etc. in java should be `appendAsString`.

add method `appendDetail(StringBuffer buffer, String fieldName, char value)` to `JsonToStringStyle`
modify `appendDetail(StringBuffer buffer, String fieldName, Object value)`

### CODE REVIEW

File: src/test/java/org/apache/commons/lang3/builder/JsonToStringStyleTest.java [hunk 1]
```diff
@@ -90,6 +90,39 @@ public void testAppendSuper() {
...
+        assertEquals("{\"a\":\"A\"}", new ToStringBuilder(base).append("a", 'A')
+                .toString());
+        assertEquals("{\"a\":\"A\",\"b\":\"B\"}", new ToStringBuilder(base).append("a", 'A').append("b", 'B')
+                .toString());
+    }
+    
+    @Test
+    public void testDate() {
+        
+        final Date now = new Date();
```
Comment (britter) [inline src/test/java/org/apache/commons/lang3/builder/JsonToStringStyleTest.java]: This fails on Travis because this import is missing.

Comment (jacktan1991) [inline src/test/java/org/apache/commons/lang3/builder/JsonToStringStyleTest.java]: Thanks!! Have fixed it.

File: src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java [hunk 2]
```diff
@@ -2568,13 +2573,13 @@ protected void appendDetail(StringBuffer buffer, String fieldName, Object value)
...
-            if (value.getClass() == String.class) {
+            if (value instanceof Number || value.getClass() == Boolean.class) {
 
-                appendValueAsString(buffer, (String)value);
+                buffer.append(value);
                 return;
             }
 
-            buffer.append(value);
+            appendValueAsString(buffer, value.toString());
```
Comment (britter) [inline src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java]: I don't think this is correct for nested object structures. Suppose we have the following class structure:

```java
public class Person {
   private String name;
   private Address address;
}

public class Address {
   private String city;
   private String zip;
}
```

Translating this to JavaScript/JSON should look like this if both, Person and Address use the JsonToStringStyle for their equals method:

```JavaScript
{
   "name": "Peter",
   "address": {
      "city": "New York",
      "zip": "10026"
   }
}
```

If we apply this fix, the result will look like this:

```JavaScript
{
   "name": "Peter",
   "address": "{
      "city": "New York",
      "zip": "10026"
   }"
}
```

Note the additional double quotes around the address object.

Comment (jacktan1991) [inline src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java]: Yes. But if `Address` like you suppose here does not *Override* `Object.toString()` (maybe someone just wants Address instance represented by its hash code), then its json style looks like `"address":"Address@1ab2f62"`.

**Comment by britter** [comment]
Hello @jacktan1991,

thank you for this contribution. I had to do some clean ups, see 07ec8ad2dd862c8669b2557b995ec5180663d73f, 28656559d2fc67193a84d5cbc48d378b2016f74a and e8db7237a9c347d2225fce9b49ff92bb741adffa. Please have a look at [CONTRIBUTING.md](CONTRIBUTING.md) if you intend to contribute on a regular basis (which would be awesome! ;-) )

**Pull request merged**